packages/quantic/force-app/main/default/lwc/quanticResultActionBar/quanticResultActionBar.js

```diff
@@ -0,0 +1,76 @@
+import {LightningElement} from 'lwc';
+
+/**
+ *  The `QuanticResultActionBar` component displays the Quantic Result Actions in a proper way with the right styles.
+ * @category Search
+ * @category Insight Panel
+ * @example
+ * <c-quantic-result-action-bar>
+ *   <c-quantic-result-action onclick={actionOne} icon-name="utility:attach" label-when-hover="Action One"></c-quantic-result-action>
+ *   <c-quantic-result-action onclick={actionTwo} icon-name="utility:email" label-when-hover="Action Two"></c-quantic-result-action>
+ *   <c-quantic-result-action onclick={actionThree} icon-name="utility:chat" label-when-hover="Action Three"></c-quantic-result-action>
+ * </c-quantic-result-action-bar>
+ */
```

Comment:
The other idea I had in mind for the design of this component was the following, did you consider it working like this instead and would it make it more simple?

```
//Template.html

<c-quantic-result-action-bar actions={actions}></c-quantic-result-action-bar>
```

```
//Template.js

this.actions = [
{ actionTrigger: (details) => {...}, icon: "utility:attach", labelWhenHover: "Action One"},
{ actionTrigger: (details) => {...}, icon: "utility:email", labelWhenHover: "Action Two"},
{ actionTrigger: (details) => {...}, icon: "utility:chat", labelWhenHover: "Action Three"},
];
```

```
//QuanticResultActionBar.html

<template>
  <template for:each={actions} for:item="act">
    <c-quantic-result-action onclick={act.actionTrigger} icon-name={act.icon} label-when-hover={act.labelWhenHover}></c-quantic-result-action>
  </template>
</template>
```

Reply:
What do you think of handling that via just a code array versus adding multiple HTML, where you need to write code for the handler function anyway?

Also the handler function of each of these actions need to know the current result they are rendering so they are able to get some result fields for example for attach, etc.

Reply:
Yes, I did consider using an array of object instead but I ended up with the slot solution because of the following reasons:
- First because the components I created are inspired by the [lightning-button-group](https://developer.salesforce.com/docs/component-library/bundle/lightning-button-group/example) component which accomplishes a very similar job as our QuanticResultActionBar component and  uses this Slots method.
- I also wanted to stay consistent with the other Quantic components that try to achieve a similar job. For example: the QuanticTabBar component, this component receives multiple QuanticTab components in its slot instead of an array of tab labels.
- Maybe more of a personal feeling, but I feel like using slots to address this kind of situation more natural than passing a whole array object property. 

What do you think? what are the pros of using an array of object property instead of using the HTML slots?
 
Any thoughts on that @nathanlb and @lbergeron ? I would like to hear your opinion as well if possible!

Reply:
I have a slight preference for the slot method for the same reasons you outlined. However the slot handlers in the actionBar component seem a little messy.